File: linear_algebra/C++/BDK_BT8.cpp
#include <iostream>

int main() {
    int n;
    std::cout << "Nhap so nguyen        : ";
    std::cin >> n;

    int ma_1[100][100];
    int unit_ma_tran[100][100];

    for (int i = 0; i < n; i++) {
        for (int j = 0; j < n; j++) {
            ma_1[i][j] = 0;
            if (i == j) {
                unit_ma_tran[i][j] = 1;
            } else {
                unit_ma_tran[i][j] = 0;
            }
        }
    }

    std::cout << "\nMa tran 0 kich thuoc " << n << "x" << n << ":\n";
    for (int i = 0; i < n; i++) {
        for (int j = 0; j < n; j++) {
            std::cout << ma_1[i][j] << " ";
        }
        std::cout << std::endl;
    }

    std::cout << "\nMa tran don vi kich thuoc " << n << "x" << n << ":\n";
    for (int i = 0; i < n; i++) {
        for (int j = 0; j < n; j++) {
            std::cout << unit_ma_tran[i][j] << " ";
        }
        std::cout << std::endl;
    }

    return 0;
}

File: linear_algebra/Python/BDK_BT8.py
n = int(input("Nhap so nguyen     nha : "))

ma_tran_0 = []
for i in range(n):
    rows = []
    for j in range(n):
        rows.append(0)
    ma_tran_0.append(rows)

unit_ma_tran = []
for i in range(n):
    rows = []
    for j in range(n):
        if i == j:
            rows.append(1)
        else:
            rows.append(0)
    unit_ma_tran.append(rows)

print(f"\nMa trankich thuoc {n}x{n}:")
for rows in ma_tran_0:
    for phan_tu in rows:
        print(phan_tu, end=" ")
    print()

print(f"\nMa tran don vi kich thuoc {n}x{n}:")
for rows in unit_ma_tran:
    for phan_tu in rows:
        print(phan_tu, end=" ")
    print()